src/main/java/org/yourteamname/utils/Data.java

```diff
@@ -0,0 +1,278 @@
+    	String bakeryName = "";
+		JSONArray products = new JSONArray();
+		
+		List<String> productType = new ArrayList();
+		List<String> productPrice = new ArrayList();
+		
+		//Get Product List
+		try {
+			for (int i = 0; i < dataArray.length(); i++) {
+				bakeryName = dataArray.getJSONObject(i).getString("name");
+				
+				if (bakeryName.equals(name)) {
+					products = dataArray.getJSONObject(i).getJSONArray("products");
+					
+					for (int j = 0; j < products.length(); j++) {
+						productType.add(products.getJSONObject(j).getString("guid"));
+						//productPrice.add(BigDecimal.valueOf(products.getJSONObject(i).getDouble("salesPrice")).floatValue());
+						productPrice.add(Double.toString(products.getJSONObject(j).getDouble("salesPrice")));
+					}
+					
+					Map<String,List<String>> map = new HashMap();
+					map.put("productType", productType);
+					map.put("productPrice", productPrice);
+					return map;
+				}
+			}
+		} catch (JSONException e) {
+			e.printStackTrace();
+		}
+		
+		return null;
+    }	
+	
+    public static JSONObject findTheCheapest(JSONObject proposal) {
+		JSONObject confirmation = new JSONObject();
+		JSONObject product = new JSONObject();
+		
+		List<String> bakeryName = new ArrayList();
+		List<String> productTypes = new ArrayList();
+		
+		String chosenBakery = "";
+		
+		//Get All Bakery Name
+		try {
+			Iterator iter = proposal.keys();
+			while(iter.hasNext()) {
+				String key = (String)iter.next();
+				bakeryName.add(key);
+				
+				product = proposal.getJSONObject(key);
+				Iterator iter2 = product.keys();
+				while(iter2.hasNext()) {
+					String key2 = (String)iter2.next();
+					if (!productTypes.contains(key2)) {
+						productTypes.add(key2);
+					}
+				}
+			}
+		} catch (JSONException e) {
+			e.printStackTrace();
+		}
+		
+		//Get The Cheapest Price
+		try {
+			for (String type : productTypes) {
+				Double min_price = Double.MAX_VALUE;
+				for (String name : bakeryName) {
+					product = proposal.getJSONObject(name);	
+					
+					if (min_price > product.getDouble(type) && product.getDouble(type) != 0) {
+						chosenBakery = name;
+						min_price = product.getDouble(type);
+					}
+				}
+				
+				if (confirmation.has(chosenBakery)) {
+					type = type + ", " + confirmation.getString(chosenBakery);
+				}
+				
+				confirmation.put(chosenBakery, type);
+			}
+		} catch (JSONException e) {
+			e.printStackTrace();
+		}
+		
+		return confirmation;
+	}
+    
+	/*
+	Orders are randomly generated. They are saved in JSON format and includes ID and ordered
+	product list. 
+	16 Nov 2018: only one order in orders. Customer ID is randomly generated to make sure 
+	there are no duplication. Number of product ordered is also randomly generated. 
+	*/ 
+	public List<JSONObject> makeOrder(List<String> product_types) {
+		List<JSONObject> orders = new ArrayList<>();
+		JSONObject order = new JSONObject();
+				
+		String customer_id = generateRandomID(10);
```

Comment:
Customers already have a guid, you should be using the one from the scenario files.

Reply:
The function is no longer used by Customer Agent. I just haven't deleted it yet. It was from last week assignment where I haven't used the scenario files yet. 

I will include all customer function to CustomerAgent.java as you suggested.